mozci/util/taskcluster.py

```diff
@@ -81,6 +83,55 @@ def find_task_id(index_path, use_proxy=False, root_url=PRODUCTION_TASKCLUSTER_RO
     return response.json()["taskId"]
 
 
+def insert_task(
+    index_path,
+    task_id,
+    rank=0,
+    expires=None,
+    data={},
+    use_proxy=False,
+    root_url=PRODUCTION_TASKCLUSTER_ROOT_URL,
+):
+    if expires is None:
+        expires = datetime.datetime.now() + datetime.timedelta(days=1 * 365)
+
+    response = _do_request(
+        get_index_url(index_path, root_url=root_url),
+        data={
+            "data": data,
+            "expires": expires,
+            "rank": rank,
+            "taskId": task_id,
+        },
+    )
+    return response.json()
+
+
+def get_indexed_tasks_url(namespace, root_url=PRODUCTION_TASKCLUSTER_ROOT_URL):
+    return liburls.api(
+        root_url,
+        "index",
+        "v1",
+        f"tasks/{namespace}",
+    )
+
+
+def list_indexed_tasks(
+    namespace, use_proxy=False, root_url=PRODUCTION_TASKCLUSTER_ROOT_URL
+):
+    url = get_indexed_tasks_url(namespace, root_url=root_url)
+    tasks = []
+    token = False
+    # Support pagination using continuation token
+    while token is not None:
+        extra_params = "?" + urlencode({"continuationToken": token}) if token else ""
+        results = _do_request(url + extra_params).json()
+        tasks += results.get("tasks")
```

Comment:
You could use `yield` here instead

Reply:
I reused some existing code from: https://github.com/mozilla/mozci/blob/master/mozci/console/commands/push.py#L915

Should I make this change anyway?

Reply:
Yes because it's a standalone method, it will be cleaner to use :

```python
yield from results.get("tasks", [])
```